src/document.ts

```diff
@@ -0,0 +1,58 @@
+import { merge } from 'lodash';
+import yaml from 'js-yaml';
+
+import type { AsyncAPIObject } from './spec-types';
+
+export interface Document {
+  _doc: AsyncAPIObject;
+}
+
+/**
+ * @class
+ *
+ * @example
+ *
+ * const document = new Document(parsedJSONList, base);
+ *
+ * console.log(document.json()); // get JSON object
+ * console.log(document.yml()); // get YAML string
+ * console.log(document.string()); // get JSON string
+ */
+
+export class Document {
+  /**
+   *
+   * @param {Object[]} parsedJSONList
+   * @param {Object} base
+   */
```

Comment:
Can we make `_doc` property as private?

```suggestion

export class Document {
  private readonly _doc: AsyncAPIObject;

  /**
   *
   * @param {Object[]} parsedJSONList
   * @param {Object} base
   */
```

And by this we can remove `export interface Document` :)